luxonis_ml/scheme/v1/config_building_blocks/base_models/head.py

```diff
@@ -0,0 +1,192 @@
+from abc import ABC
+from typing import List, Optional, Union
+
+from pydantic import BaseModel, Field, validator
+
+from ..enums import ObjectDetectionSubtypeYOLO
+from .custom_base_model import CustomBaseModel
+
+
+class HeadMetadata(BaseModel, ABC):
+    """Head metadata parent class.
+
+    @type family: str
+    @ivar family: Decoding family.
+    @type classes: list
+    @ivar classes: Names of object classes recognized by the model.
+    @type n_classes: int
+    @ivar n_classes: Number of object classes recognized by the model.
+    """
+
+    family: str = Field(description="Decoding family.")
+    classes: List[str] = Field(
+        description="Names of object classes recognized by the model."
+    )
+    n_classes: int = Field(
+        description="Number of object classes recognized by the model."
+    )
+
+
+class HeadMetadataClassification(HeadMetadata):
+    """Metadata for classification head.
+
+    @type family: str
+    @ivar family: Decoding family.
+    @type is_softmax: bool
+    @ivar is_softmax: True, if output is already softmaxed.
+    """
+
+    family: str = Field("Classification", Literal=True, description="Decoding family.")
```

Comment:
`Literal` is not an argument for `Field`. Is that supposed to go to extra JSON fields? If so, it should be included in the `json_schema_extra` argument instead:

```python
family: str = Field("Classification", description="Decoding family.", json_schema_extra={"Literal": True})
```